crates/ika-core/src/dwallet_mpc/crytographic_computation/mpc_computations/presign.rs

```diff
@@ -191,7 +200,15 @@ impl PresignPublicInputByProtocol {
                     .secp256k1_protocol_public_parameters();
 
                 let deserialized_dwallet_public_output =
-                    dwallet_public_output.and_then(|bytes| bcs::from_bytes(&bytes).ok());
+                    dwallet_public_output.and_then(|bytes| {
+                        match bcs::from_bytes::<<Secp256K1ECDSAProtocol as dkg::Protocol>::DecentralizedPartyDKGOutput>(&bytes) {
+                            Ok(output) => Some(output.into()),
+                            Err(_) => {
+                                warn!(session_identifier=?session_identifier, "Failed to deserialize dwallet public output for session.");
```

Comment:
No, not only log warn - you should actually return an error and reject the session